share/smack/lib/smack.c

```diff
@@ -937,10 +937,13 @@ void __SMACK_decls() {
 
 #if MEMORY_MODEL_NO_REUSE_IMPLS
   D("var $Alloc: [ref] bool;");
+  D("function $Size(ref) returns (ref);");
   D("var $CurrAddr:ref;");
 
   D("procedure $alloc(n: ref) returns (p: ref)\n"
     "modifies $CurrAddr, $Alloc;\n"
+    "ensures $Size(p) == n;\n"
```

Comment:
We do not need these ensures, we only need the assumes below. So please remove them.